harness/determined/estimator/_callback.py

```diff
@@ -28,3 +28,9 @@ def on_checkpoint_end(self, checkpoint_dir: str) -> None:
             only on the chief GPU (rank = 0) which performs the checkpoint.
         """
         pass
+
+    def on_training_end(self) -> None:
+        """
+        Run when training completes successfully.
```

Comment:
blocking: Probably add a comment here to tell users this is the place they should close/clean up their data loaders.

Reply:
Updated comment, also renamed to `on_trial_close()`.